beacon/blockchain/process_proposal.go

```diff
@@ -72,56 +78,47 @@ func (s *Service[
 		)
 	if err != nil {
 		return createProcessProposalResponse(errors.WrapNonFatal(err))
+	} else if sidecars.IsNil() {
+		s.logger.Warn(
+			"Aborting block verification - blob sidecars not found in proposal",
+		)
+		return createProcessProposalResponse(errors.WrapNonFatal(ErrNilBlob))
 	}
 
-	// In theory, swapping the order of verification between the sidecars
-	// and the incoming block should not introduce any inconsistencies
-	// in the state on which the sidecar verification depends on (notably
-	// the currently active fork). ProcessProposal should only
-	// keep the state changes as candidates (which is what we do in
-	// VerifyIncomingBlock).
+	// Make sure we have the right number of BlobSidecars
+	numCommitments := len(blk.GetBody().GetBlobKzgCommitments())
+	if numCommitments != sidecars.Len() {
```

Comment:
although we already check before calling `.Len()` if `sidecars.IsNil()` I think we should prevent any `sidecars` function from ever panicking on its own

Reply:
i.e. here [Len, GetSidecars, Get](https://github.com/berachain/beacon-kit/pull/2291/files#diff-29cacb221d7916f7d9d44e53bc63db6dc41f5e5f816f9634eb5faf5808e98417R38-R48) should all check .IsNil before dereferencing the pointer.

Reply:
we do in line 81 right?